hawc/apps/vocab/constants.py

```diff
@@ -12,25 +11,25 @@ class VocabularyNamespace(IntegerChoices):
     EHV = 1, "EHV"
     ToxRefDB = 2, "ToxRefDB"
 
-    @classproperty
-    def display_dict(cls) -> dict:
-        return {1: "EPA Environmental health vocabulary", 2: "EPA ToxRefDB vocabulary"}
-
-    @classproperty
-    def display_urls(cls) -> dict:
-        return {1: reverse_lazy("vocab:ehv-browse"), 2: reverse_lazy("vocab:toxrefdb-browse")}
-
     @classmethod
     def display_choices(cls) -> list:
-        return [(key, value) for key, value in cls.display_dict.items()]
+        return [(item, item.display_name) for item in cls]
 
     @property
     def display_url(self) -> str:
-        return self.display_urls[self.value].lower()
+        match self:
```

Comment:
I switched to a match case instead of two methods for each; I suppose it may not be quite as speedy, but it should be plenty fast either way